engine/websocketroutine_manager.go

```diff
@@ -159,6 +162,12 @@ func (m *websocketRoutineManager) WebsocketDataHandler(exchName string, data int
 			exchName)
 	}
 
+	m.mu.RLock()
```

Comment:
Unrelated to your change, but in the function `WebsocketDataReceiver(ws *stream.Websocket) {`, we go: `m.wg.Add(1)`, could you instead add this before the go function is called inside `websocketRoutine`? eg:
```go
				if ws.IsEnabled() {
					err = ws.Connect()
					if err != nil {
						log.Errorf(log.WebsocketMgr, "%v", err)
					}
					m.wg.Add(1)
					go m.WebsocketDataReceiver(ws)
					err = ws.FlushChannels()
					if err != nil {
						log.Errorf(log.WebsocketMgr, "Failed to subscribe: %v", err)
					}
				}
```